sdks/go/pkg/beam/io/rtrackers/offsetrange/offsetrange_test.go

```diff
@@ -63,10 +63,87 @@ func TestRestriction_EvenSplits(t *testing.T) {
 					t.Errorf("split restriction [%v, %v] has unexpected size. got: %v, want: %v or %v",
 						split.Start, split.End, size, min, min+1)
 				}
-				// Check: All elements are still in a split restrictions. This
-				// logic assumes that the splits are returned in order which
-				// isn't guaranteed by EvenSplits, but this check is way easier
-				// with the assumption.
+				// Check: All elements are still in a split restriction and
+				// the restrictions are in the appropriate ascending order.
+				if split.Start != prevEnd {
+					t.Errorf("restriction range [%v, %v] missing after splits.",
+						prevEnd, split.Start)
+				} else {
+					prevEnd = split.End
+				}
+			}
+			if prevEnd != r.End {
+				t.Errorf("restriction range [%v, %v] missing after splits.",
+					prevEnd, r.End)
+			}
+		})
+	}
+}
+
+// TestRestriction_SizedSplits tests various splits and checks that they all
+// follow the contract for SizedSplits. This means that all restrictions match
+// the given size unless it is a remainder, and that each element is present
+// in the split restrictions.
+func TestRestriction_SizedSplits(t *testing.T) {
+	tests := []struct {
+		rest Restriction
+		size int64
+		want []Restriction
+	}{
+		{
+			rest: Restriction{Start: 0, End: 11},
+			size: 5,
+			want: []Restriction{{0, 5}, {5, 10}, {10, 11}},
+		},
+		{
```

Comment:
Consider adding an exact case too eg.
Start 7, End 17
size 5
{7, 12}, {12, 17}
IIUC the implementation correctly.
If I understand the

Reply:
Yeah, that seems like a good idea. I'll add one more case before merging.